src/utils/configurations/useSetUpConfigurations.js

```diff
@@ -0,0 +1,65 @@
+import { useDataEngine } from '@dhis2/app-runtime'
+import { useState, useEffect } from 'react'
+import defaultConfigurations from './defaultConfigurations.json'
+
+const DATASTORE_KEY =
+    process.env.REACT_APP_DHIS2_APP_DATASTORE_KEY || 'configurations'
+export const DATASTORE_ENDPOINT = 'dataStore/who-dqa/' + DATASTORE_KEY
+
+const CONFIGURATIONS_QUERY = {
+    configurations: {
+        resource: DATASTORE_ENDPOINT,
+    },
+}
+const SET_UP_CONFIGURATIONS_MUTATION = {
+    resource: DATASTORE_ENDPOINT,
+    type: 'create',
+    data: defaultConfigurations,
+}
+
+export const useSetUpConfigurations = (setConfigurations) => {
+    const engine = useDataEngine()
+    const [loading, setLoading] = useState(true)
+    const [error, setError] = useState(false)
+
+    useEffect(() => {
+        engine
+            // attempt to fetch configurations
+            .query(CONFIGURATIONS_QUERY)
+            // return configurations object to next handlers
+            .then((data) => data.configurations)
+            .catch((err) => {
+                // if the fetch failed, we might need to set up the data store
+                const { details } = err
+                const needToSetUpDatastore =
+                    details?.httpStatusCode === 404 &&
+                    /Key .* not found in namespace/.test(details?.message)
```

Comment:
I'm not sure we want to test for the specific message as this could theoretically change. It seems sufficient to just check the status code and confirm that it's 404